pkg/operator/controller/controller_dns_service.go

```diff
@@ -93,12 +93,19 @@ func serviceChanged(current, expected *corev1.Service) (bool, *corev1.Service) {
 		cmpopts.IgnoreFields(corev1.ServiceSpec{}, "ClusterIP", "TopologyKeys"),
 		cmpopts.EquateEmpty(),
 	}
-	if cmp.Equal(current.Spec, expected.Spec, serviceCmpOpts...) {
+
+	servingCertAnnotationKey := "service.beta.openshift.io/serving-cert-secret-name"
+	currentServingCertAnnotation := current.ObjectMeta.Annotations[servingCertAnnotationKey]
+	expectedServingCertAnnotation := expected.ObjectMeta.Annotations[servingCertAnnotationKey]
+	annotationMatches := cmp.Equal(currentServingCertAnnotation, expectedServingCertAnnotation)
```

Comment:
As you are comparing just the single annotation's value, `cmp.Equal` should be unnecessary; `annotationMatches := currentServingCertAnnotation == expectedServingCertAnnotation` should suffice.